models/classes/Lists/Business/Service/RemoteSource.php

```diff
@@ -45,13 +46,32 @@ public function __construct(array $parsers, Client $client = null)
         $this->client  = $client;
     }
 
+    /**
+     * @deprecated Use $this->fetchContext()
+     */
     public function fetch(string $sourceUrl, string $uriPath, string $labelPath, string $parser): Traversable
     {
-        $response = $this->getClient()->get($sourceUrl);
+        $context = new RemoteSourceContext([
+            RemoteSourceContext::PARAM_SOURCE_URL => $sourceUrl,
+            RemoteSourceContext::PARAM_URI_PATH => $uriPath,
+            RemoteSourceContext::PARAM_LABEL_PATH => $labelPath,
+            RemoteSourceContext::PARAM_PARSER => $parser,
+        ]);
+
+        yield from $this->fetchContext($context);
+    }
 
-        $body = json_decode((string)$response->getBody(), true);
+    public function fetchContext(RemoteSourceContext $context): Traversable
```

Comment:
👏 👏 👏  🚀

Reply:
Cause you are fetching the resources by context, not fetching the context. Wdyt?

```suggestion
    public function fetchByContext(RemoteSourceContext $context): Traversable
```